modules/standard/Socket.md

```diff
@@ -0,0 +1,210 @@
+  will return a reader channel for associated socket
+  ```python
+  proc tcpConn.writer() : channel
+  ```
+- `writer`
+  will return a writer channel for associated socket
+  ```python
+  proc tcpConn.writer() : channel
+  ```
+- `close` calling this on `tcpConn` will close the socket connection and associated `file` too.
+  ```python
+  proc tcpConn.close()
+  ```
+- `addr` this function will return a `ipAddr` object which will contain the `host` and `port` where socket is bound.
+
+## Creating a Listener
+
+`listen` function will create a `tcpListener` Instance which will take in following parameters
+
+**Method:**
+```python
+proc listen(addr:ipAddr, backlog=5, reuseaddr=true) : tcpListener
+```
+
+**Parameters**
+```
+addr: ipAddr - contains family info of socket to create and where to bind it.
+backlog: int - optional parameter, default value = 0
+reuseaddr: Boolean - optional parameter, default value = true
+```
+
+- The `backlog` argument defines the maximum length to which specifies the queue length for completely established sockets waiting to be accepted. If a connection request arrives when the queue is full, the client may receive an error with an indication of `ECONNREFUSED` or, if the underlying protocol supports retransmission, the request may be ignored so that a later reattempt at connection succeeds.
+- Calling in `listen` without `reuseaddr` can return an error that some other socket is already bound to the specified address. This can also be a result of re-running the program after only a short waiting period where the `bind` may find address as already in use.
+
+The `tcpListener` instance will have methods on it to call
+- `accept`: method will return a `tcpConn` instance of any new incoming connection otherwise will yield if no new connection is available at the instant. The communication can then proceed through the `tcpConn` instance returned
+  ```python
+  proc tcpListener.accept() : tcpConn
+  ```
+- `close`,
+  calling close on `tcpListener` will close the socket server and it won't accept anymore new connections.
+  ```python
+  proc tcpListener.close()
+  ```
+- `family` returns the family of socket `AF_INET` or `AF_INET6`
+- `addr` will return an `ipAddr` instance providing details about the address and port at which socket is listening.
+  ```python
+  proc tcpListener.addr() : ipAddr
+  ```
+
+The additional support with `tcpConn` and `tcpListener` will be to enable or disable two of the prominent optimization algorithm for TCP Protocol that are:
+
+- `nagle`
+  This method will set/unset TCP_NODELAY for provided socket.
+  ```python
+  proc nagle(socket:tcpConn, tcpNodeDelay:bool = 1)
+  proc nagle(socket:tcpListener, tcpNodeDelay:bool = 1);
+  proc nagle(socket:udpSocket, tcpNodeDelay:bool = 1);
+  ```
+- `quickack`
+  This method will set/unset TCP_QUICKACK for provided socket.
+  ```python
+  proc quickack(socket:tcpConn, tcpQuickack:bool = 1)
+  proc qucikack(socket:tcpListener, tcpQuickack:bool = 1);
```

Comment:
Typo.

More importantly, as someone not so knowledgeable about this matters, I read "quickack" as two separate words, and left to my own devices would capitalize as `quickAck`

Reply:
- Correcting the typo
- This was based on Julia's [quickack](https://docs.julialang.org/en/v1/stdlib/Sockets/#Sockets.quickack) function they seem to use lowercase naming for all so am also not sure if it's actually two words but seems so.

Reply:
Apparently, their convention is "put multiple words together if they're readable, snake_case if they aren't", where I believe we are leaning more aggressively towards distinguishing words. So, I still personally lean towards quickAck, but I'd prefer others to speak up about this before making any change.

Relevant part of julia naming conventions: https://docs.julialang.org/en/v1/manual/style-guide/#Use-naming-conventions-consistent-with-Julia-base/

Reply:
I'd also be inclined to use `quickAck` in this case.  It doesn't feel sufficiently compound / single-word to meet my personal "all lowercase" test.

Reply:
made the change 😄

Reply:
There is one particular change that I did like to take opinions on. I didn't check it earlier but the actual optimizations are called `naggle` and `delayAck` which are enabled on disabling flag `TCP_NODEDELAY` and `TCP_QUICKACK` respectively. 

The Julia idea of naming one procedure on optimization and the other on the flag doesn't seem right. We can make changes either to name both based on `flags` they actually change or `optimization` they enable which can be a bit confusing since to enable them we have to disable something else but it might be more known.